edweiss-firebase/functions/src/functions/lectures/clearQuestionEvent.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * @file clearQuestionEvents.ts
+ * @description Cloud function for removing the current question lecture event 
+ * @author Adamm Alaoui
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import LectureDisplay from 'model/lectures/lectureDoc';
+import { onAuthentifiedCall } from 'utils/firebase';
+import { CollectionOf, getDocumentAndRef } from 'utils/firestore';
+import { fail, ok } from 'utils/status';
+
+
+// Types
+type Lecture = LectureDisplay.Lecture;
+
+// ------------------------------------------------------------
+// -----------     Clear Question Cloud Function      ---------
+// ------------------------------------------------------------
+
+export const clearQuestionEvent = onAuthentifiedCall(LectureDisplay.Functions.clearQuestionEvent, async (userId, args) => {
+    if (!args.courseId || !args.lectureId) {
+        return fail('invalid_arg');
+    }
+
+    const [lecture, lectureRef] = await getDocumentAndRef(CollectionOf<Lecture>(`courses/${args.courseId}/lectures`), args.lectureId);
+
+    if (lecture == undefined)
+        return fail("error_firebase");
+
+
+    try {
+        await lectureRef.update({
+            ["event"]: {
+                id: "",
+                type: "invalid",
+            }
+        });
+    } catch (error) {
+        return fail('error_firebase');
+    }
+
+
+    return ok({ id: args.id });
+});
```

Comment:
Why do you return id ?

Reply:
No particular reasons. It just for debugging purposes that I did it. So that in the `res` value I could determine grab the `id` value to check that the firebase function has been correctly called. Can we keep it this way ? An empty one would also o the trick, I aggree with you on this point.

Reply:
Maybe at the and of the project yon can remove it for understandability ?

Reply:
Yes of course, look into it for next sprint.